### PR TITLE
Add API endpoint for filtering messages in the message store

### PR DESCRIPTION
**Pull request by ermyas**
This PR adds an API endpoint for filtering messages based on event details such as `blockNumber`, `txIndex` etc. Currently, this capability is enabled through a dedicated `/events/...` endpoint. However, this is approach is somewhat confusing and limiting, because the message store does not maintain raw events and the endpoint itself returns message objects instead of events. This PR makes the following changes to address this:
1) adds an ability to query the `/messages` root endpoint with relevant parameters
2) removes the `/events/...` endpoint

### CODE REVIEW

**Comment by drinkcoffee** [comment]
I will leave this one for @wcgcyx  to review

File: docs/spec/message-store-service-api.yml [hunk 1]
```diff
@@ -11,22 +11,97 @@ tags:
...
+          in: query
+          description: ID of network in which the event was generated
+          required: true
+          type: string
+        - name: blockNumber
+          in: query
+          description: The block number that contains this event
+          required: true
+          type: string
+        - name: txIndex
```
Comment (wcgcyx) [inline docs/spec/message-store-service-api.yml:38]: How about adding filter message by transaction hash? Might be easier for user to query

**Review by wcgcyx — APPROVED** [review]
Approved, left a comment

File: docs/spec/message-store-service-api.yml (hunk 1 above)
Comment (ermyas) [reply docs/spec/message-store-service-api.yml:38]: sorry for the delayed response @wcgcyx
Yes, transaction hash makes sense, I'll add it

**Pull request merged**